Userland/Applications/Browser/CookieJar.cpp

```diff
@@ -287,93 +335,251 @@ void CookieJar::store_cookie(Web::Cookie::ParsedCookie const& parsed_cookie, con
+            if (cookie_path_length > entry.path.length()) {
                 return true;
-            } else if (cookie.value.path.length() == entry.path.length()) {
-                if (cookie.value.creation_time.timestamp() < entry.creation_time.timestamp())
+            } else if (cookie_path_length == entry.path.length()) {
+                if (cookie_creation_time < entry.creation_time.timestamp())
                     return true;
             }
             return false;
         });
+    });
+
+    // 3. Update the last-access-time of each cookie in the cookie-list to the current date and time.
+    auto now = Core::DateTime::now();
 
-        // 3. Update the last-access-time of each cookie in the cookie-list to the current date and time.
-        cookie.value.last_access_time = now;
+    for (auto& cookie : cookie_list) {
+        cookie.last_access_time = now;
+        update_cookie_in_database(cookie);
     }
 
     return cookie_list;
 }
 
-void CookieJar::purge_expired_cookies()
+static ErrorOr<Web::Cookie::Cookie> parse_cookie_from_database(Span<SQL::Value const> row)
 {
-    time_t now = Core::DateTime::now().timestamp();
-    Vector<CookieStorageKey> keys_to_evict;
+    if (row.size() != 12)
+        return Error::from_string_view("Incorrect number of columns to parse cookie"sv);
+
+    size_t index = 0;
+
+    auto convert_text = [&](auto& field, StringView name) -> ErrorOr<void> {
+        auto const& value = row[index++];
+        if (value.type() != SQL::SQLType::Text)
+            return Error::from_string_view(name);
+
+        field = value.to_deprecated_string();
+        return {};
+    };
+
+    auto convert_bool = [&](auto& field, StringView name) -> ErrorOr<void> {
+        auto const& value = row[index++];
+        if (value.type() != SQL::SQLType::Boolean)
+            return Error::from_string_view(name);
+
+        field = value.to_bool().value();
+        return {};
+    };
+
+    auto convert_time = [&](auto& field, StringView name) -> ErrorOr<void> {
+        auto const& value = row[index++];
+        if (value.type() != SQL::SQLType::Float)
+            return Error::from_string_view(name);
 
-    for (auto const& cookie : m_cookies) {
-        if (cookie.value.expiry_time.timestamp() < now)
-            keys_to_evict.append(cookie.key);
+        // FIXME: Support 64-bit types in LibSQL::Value.
+        auto time = static_cast<time_t>(value.to_double().value());
+        field = Core::DateTime::from_timestamp(time);
+        return {};
+    };
+
+    auto convert_same_site = [&](auto& field, StringView name) -> ErrorOr<void> {
+        auto const& value = row[index++];
+        if (value.type() != SQL::SQLType::Integer)
+            return Error::from_string_view(name);
+
+        auto same_site = value.to_int().value();
+        if (same_site > to_underlying(Web::Cookie::SameSite::Lax))
+            return Error::from_string_view(name);
```

Comment:
These errors are gonna be pretty annoying to deal with, "Error: same_site"...

Reply:
Kinda just dropping the errors on the floor right now, should probably be logging them though

Reply:
We'll log the attribute and it's value now:
```
109.320 Browser(58:58): Failed to parse cookie attribute 'same_site': 12
```